test/powershell/Modules/Microsoft.PowerShell.Utility/Sort-Object.Tests.ps1

```diff
@@ -21,204 +21,946 @@
+            $topNSortResults[$i] | Should Be $fullSortResults[$i]
+        }
+    }
+
+    It 'Return the bottom N objects sorted in descending order using two properties' {
+        $unsortedData = @(
+            $item0 = [pscustomobject]@{FirstName='John'  ;LastName='Smith';YearsInMS=5 }
+            $item1 = [pscustomobject]@{FirstName='Joseph';LastName='Smith';YearsInMS=15}
+            $item2 = [pscustomobject]@{FirstName='John'  ;LastName='Smyth';YearsInMS=12}
+            $item3 = [pscustomobject]@{FirstName='Johnny';LastName='smith';YearsInMS=8 }
+        )
+        $baseSortParameters = @{Property='LastName','FirstName';Descending=$true}
+        $fullSortResults = $unsortedData | Sort-Object @baseSortParameters
+        $bottomNSortResults = $unsortedData | Sort-Object @baseSortParameters -Bottom 2
+        $offset = $fullSortResults.Count - $bottomNSortResults.Count
+        for ($i = 0; $i -lt $bottomNSortResults.Count; $i++) {
+            $bottomNSortResults[$i] | Should Be $fullSortResults[$i + $offset]
+        }
+    }
+
+    It 'Return the top N objects using two properties with mixed sort order' {
+        $unsortedData = @(
+            $item0 = [pscustomobject]@{FirstName='John'  ;LastName='Smith';YearsInMS=5 }
+            $item1 = [pscustomobject]@{FirstName='Joseph';LastName='Smith';YearsInMS=15}
+            $item2 = [pscustomobject]@{FirstName='John'  ;LastName='Smyth';YearsInMS=12}
+            $item3 = [pscustomobject]@{FirstName='Johnny';LastName='smith';YearsInMS=8 }
+        )
+        $baseSortParameters = @{Property=@{Expression='FirstName';Ascending=$true},@{Expression='LastName'};Descending=$true}
+        $fullSortResults = $unsortedData | Sort-Object @baseSortParameters
+        $topNSortResults = $unsortedData | Sort-Object @baseSortParameters -Top 2
+        for ($i = 0; $i -lt $topNSortResults.Count; $i++) {
+            $topNSortResults[$i] | Should Be $fullSortResults[$i]
+        }
+    }
+
+    It 'Return the bottom N objects using two properties with mixed sort order' {
+        $unsortedData = @(
+            $item0 = [pscustomobject]@{FirstName='John'  ;LastName='Smith';YearsInMS=5 }
+            $item1 = [pscustomobject]@{FirstName='Joseph';LastName='Smith';YearsInMS=15}
+            $item2 = [pscustomobject]@{FirstName='John'  ;LastName='Smyth';YearsInMS=12}
+            $item3 = [pscustomobject]@{FirstName='Johnny';LastName='smith';YearsInMS=8 }
+        )
+        $baseSortParameters = @{Property=@{Expression='FirstName';Ascending=$true},@{Expression='LastName'};Descending=$true}
+        $fullSortResults = $unsortedData | Sort-Object @baseSortParameters
+        $bottomNSortResults = $unsortedData | Sort-Object @baseSortParameters -Bottom 2
+        $offset = $fullSortResults.Count - $bottomNSortResults.Count
+        for ($i = 0; $i -lt $bottomNSortResults.Count; $i++) {
+            $bottomNSortResults[$i] | Should Be $fullSortResults[$i + $offset]
+        }
+    }
+
+}
```

Comment:
http://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline